packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -24,6 +24,7 @@ import {
 
 declare module 'apollo-server-env/dist/fetch' {
   interface RequestInit {
+    cacheKey?: string;
```

Comment:
I realize you're simply propagating an existing pattern, but it's pretty odd to me that we augment `RequestInit` **globally** with `cacheOptions` (and with this, also `cacheKey`).

Could you consider / look into another way of accomplishing this?  We use `RequestInit` all over the place in this repository and I'm pretty sure not all of them offer `HTTPCache` options.

Do you think another option here would be allowing a pluggable `fetch` implementation (like we do in other places, e.g., `@apollo/gateway`) is a better direction?  Specifically, [`make-fetch-happen`](https://www.npmjs.com/package/make-fetch-happen) comes with a great HTTP cache implementation. (And yes, yhis comes with the suggestion that we removing the `HTTPCache` entirely.)

Reply:
I'd be up for a cache-free RESTDataSource, to be honest :) I already control the cache for the rest of data-sources. Having an opt-in pluggable caching solution would be preferable.

We are finishing up a crunch at work so I think early next week I could have some time to have a look.

The only issue I see is that that change would be much larger than adding the cacheKey override and could be not backwards-compatible.

I guess we could have an option to add the current cache as a default that could be turned off on object instantiation.